src/app/Shared/Services/Api.service.tsx

```diff
@@ -587,4 +636,9 @@ export interface RecordingAttributes {
   events: string;
   duration?: number;
   options?: RecordingOptions;
+  labels?: string;
```

Comment:
This field addition doesn't seem quite right - shouldn't it be a `Metadata` type to better reflect the actual API response objects?

Also, the metadata field is present in the API responses for both active and archived recordings, right? I think in that case the field should be added to the `ArchivedRecording` base interface rather than here.

Reply:
Are `RecordingAttribute`s used elsewhere besides passing the Create Recording fields to `createRecording(recordingAttributes: RecordingAttributes): Observable<boolean>`? 
I originally wanted to pass `recordingAttributes.labels` to `createRecording`, which I can change to `recordingAttributes.metadata`. I've already added the `metadata: Metadata` field in the `ArchivedRecording` interface.

Reply:
It's only used in the `CreateRecording` and `CustomRecordingForm`. It just seems that it would be easier to reason about and maintain this code if the labels map is preserved as a map when passed around, rather than being stringified here - similar reasoning to what we discussed on the backend.

Reply:
WDYT about passing the `Metadata` object around in my last commit, where the object only gets stringified at `form.append('metadata', JSON.stringify(recordingAttributes.metadata))`?

Reply:
Looks better, thanks